dispatch/apps/events/sources.py

```diff
@@ -104,13 +104,29 @@ def get_data(self):
 
         html = requests.get(self.url).text
         soup = BeautifulSoup(html, 'html.parser')
-        group = soup.find_all('td', class_='fieldval')
+        sub_fields = soup.find_all('td', class_='fieldval')
+
+        title = soup.find('table', id='eventTitle').find('h2', class_='bwStatusConfirmed').find('a').text
+
+        time_string = str(sub_fields[0].text)
+        time_strings = map(str.strip, time_string.split('-'))
+
+        try:
+            start_time = datetime.strptime(time_strings[0], '%A, %B %d, %Y %I:%M %p')
+            end_time = datetime.strptime(time_strings[1], '%A, %B %d, %Y %I:%M %p')
+        except ValueError:
+            start_time = datetime.strptime(time_strings[0], '%A, %B %d, %Y %I:%M %p')
+            date = start_time.date()
+            time = datetime.strptime(time_strings[1], '%I:%M %p').time()
+            end_time = datetime.combine(date, time)
```

Comment:
I'd just like to explain this snippet. There are two forms that UBCevents Calendar uses for their start and end times: if it is a same day event, it has the form of `Saturday, August 12, 2017 9:00 AM - 1:00 PM`, and if it is multi-day, `Saturday, August 12, 2017 9:00 AM - Sunday, August 13, 2017 1:00 PM`. I first check for the multi-day case, and if that fails (i.e. when you can't parse an entire datetime from just a time), it combines the date of the `start_time` and the time of the `end_time`.